### DISCUSSION TITLE
「分享」快捷指令，支持多标签、剪贴板、多种内容分享到Memos

### DISCUSSION THREAD

**Discussion by loudtell** [usememos/memos#651] [Ideas]
## 前言
参考了官方讨论区 @monlor [分享的快捷指令](https://github.com/usememos/memos/discussions/52)，花了一天的时间进行了彻底的重写。加强了指令的功能，基本上满足了各种分享和记录到 Memos 的功能。

技术能力有限，代码不漂亮，也不够健壮，个人使用的时候要注意按照提示使用。

## 版本发布

⚠️注意：快捷指令只在iOS16和iOS15（ 经@lclrc 测试）下进行过测试，可能包含其他iOS版本不支持的代码，请自行测试。

### 20240925 更新版
⬇️[点击获取指令](https://www.icloud.com/shortcuts/e03a0e8a4f214557b90a230b1762f879)

更新内容：

- 修复了由于文件和图片转换导致的图片无法上传的问题

### 20240921 更新版
⬇️[点击获取指令](https://www.icloud.com/shortcuts/7473dbeef25c41a48126156adf908cd4)

希望以后官方的API不要再变了，用平板改这种代码多的快捷指令实在是太卡了。

更新内容：

- 使用 使用新版API重写，可以适配v0.22版本
- 删除部分之前不用的代码

### 20240121 更新版

⬇️[点击获取指令](https://www.icloud.com/shortcuts/76c7531acf05406dbc20733b04b75552)

更新内容：

- 使用 V1 版 API
- 删除部分之前不用的代码
- 调整了资源列表的实现

### 20230413更新版（来自于 @Wujiao233 三周前修改🙏）
⬇️[点击获取指令](https://www.icloud.com/shortcuts/5038b32bc3ed42b390075f6f2ede7032)
#### 更新内容：
- 更换了上传文件的API地址。解决文件分享时上传失败问题。

### 20221206更新版
⬇️[点击获取指令](https://www.icloud.com/shortcuts/474d472bf62a4a92af45044812f28990)
#### 更新内容：
- 增加可以分享联系人到Memos的功能
- 增加一个开关，控制是否提示公开笔记，开启后每次发布内容，会提示公开程度
- 增加一个开关，控制通过分享获取的内容是否提示输入笔记，开启后每次分享内容会弹出输入框要求输入笔记。
- 增加了一个默认可见度的设置，如果不想每次选择可见度，也可以将默认可见度设置成自己想要的

### 20221203更新版：
⬇️[点击获取指令](https://www.icloud.com/shortcuts/e05f7c4516c04b0d91f0730222c82e73)
#### 更新内容：
- 修改标签支持保留选择的项目并增加手动输入内容
- 新增支持自动获取剪贴板文本内容到输入框
- 新增支持获取剪贴板中的文件和图片并提示是否将其上传到Memos
- 优化流程，让整个流程更清晰

### 20221202版 ：
⬇️[点击获取指令](https://www.icloud.com/shortcuts/47539ab5e6f14a6786e969b3224d6430)

#### 功能特点：

- 支持点开指令录入内容
- 支持多标签选择和录入
- 支持不使用任何标签
- 支持从各种应用分享的以下类型内容
  - 文本
  - Safari 网页：获取标题和 URL，如果当前页面有已选择高亮的内容则也会获取到内容中。
  - URL：只获取 URL 地址，因为同时获取标题偶尔会导致出现两次地址
  - 图片：可以同时分享多个图片到 Memos
  - 文件：可以同时分享多个文件和图片到 Memos
  - 综合内容：App 分享的内容同时包含图片、文本和 URL，也会全部提交到 Memos

**Comment by lhuanyun** [comment]
系统版本多少呢？我ios13使用上存在问题。一部分步骤没有执行

**Reply by loudtell** [reply]
ios16 😂

**Reply by fastxteam** [reply]
我memos设置的是默认所有人可见，但是通过快捷指令上传的内容还要自己手动去放开所有人可见，建议放开，这样我想共享什么内容直接就可以实时共享在最前端了

**Reply by lhuanyun** [reply]
我是ios13，我觉得这样的逻辑挺好的。 https://user-images.githubusercontent.com/20707383/205668863-07751d51-5b93-4b0e-b856-f271c8aa6857.MOV

**Reply by loudtell** [reply]
> 我memos设置的是默认所有人可见，但是通过快捷指令上传的内容还要自己手动去放开所有人可见，建议放开，这样我想共享什么内容直接就可以实时共享在最前端了

感谢你的反馈，今天或者明天我会抽空更新一个版本，允许在开始的时候设置一个开关，控制是否提示公开笔记。

**Comment by loudtell** [comment]
@lhuanyun 感谢建议。

- 我个人认为应该要尽量减少选择和提示次数。
- 晚上我会再仔细看看视频，如果有需要的功能我会增加一个开关，可以自行设置是否开启相关选项。

**Comment by loudtell** [comment]
@AssistantLiu 快捷指令升级了一个新的版本，欢迎测试，更新内容已经放到主帖中了。

**Comment by Wujiao233** [comment]
更新修复了图片&文件上传的问题： 
https://www.icloud.com/shortcuts/5038b32bc3ed42b390075f6f2ede7032

**Reply by lclrc** [reply]
赞 试了下文件上传可以

**Reply by loudtell** [reply]
好久没用过分享上传的功能了，没想到接口变了🤔感谢🙏。我把这个直接分享到主帖上了

**Comment by lclrc** [comment]
谢谢分享! iOS15可以!

**Comment by nageshuizeii** [comment]


**Reply by nageshuizeii** [reply]
- [x] 原来是自己设置错误了，down

**Comment by Wujiao233** [comment]
支持0.14版本的API修改（没更新前不需要换）
https://www.icloud.com/shortcuts/90bbba8249c04db28829528e80c525ef

**Comment by gygy** [comment]
ios 17提示错误
![image](https://github.com/usememos/memos/assets/2096830/adf855a4-fd71-42b2-b17b-23c595ce3e18)

**Reply by loudtell** [reply]
应该是API又换了，稍等等吧，我这几天抽空研究研究
PS：刚看了一下网页版都无法显示tags了，应该是这次升级导致了什么错误。先等官方更新吧。

**Reply by gygy** [reply]
测试了0.21 版本，快捷指令和上面一样的提示错误

**Reply by loudtell** [reply]
你重新获取一下快捷指令，我这测试没有问题。

**Reply by gygy** [reply]
> 你重新获取一下快捷指令，我这测试没有问题。

ios 版本17.1.1 ，重新下载指令，还是报截图的错误

**Comment by gygy** [comment]
> 应该是API又换了，稍等等吧，我这几天抽空研究研究 PS：刚看了一下网页版都无法显示tags了，应该是这次升级导致了什么错误。先等官方更新吧。

0.22 版本刚刚出来，我的也有问题。

**Reply by LightUp-Tao** [reply]
0.22版本出来，遇到问题了。网页版无法显示tags，可以点击重新构建tags解决。

**Reply by LightUp-Tao** [reply]
但是快捷输入就显示无法构建字典。

**Reply by lxy12300ab** [reply]
> 0.22版本出来，遇到问题了。网页版无法显示tags，可以点击重新构建tags解决。

请问如何操作来重新构建tag，目前我是手工编辑，但是数据太多了

**Comment by lxy12300ab** [comment]
期待快捷指令更新到最新的版本

**Reply by tabortao** [reply]
啥时候更新下0.22版本的快捷指令？

**Reply by ZhXWei** [reply]
不能使用了，更新下最新版吧

**Reply by loudtell** [reply]
已经更新了

**Reply by Cr3stro** [reply]
iOS17.3.1无法通过快捷指令上传照片（通过相册分享到快捷指令），提示图片上传失败是否继续，请问有无微调修复的方法

**Reply by loudtell** [reply]
> iOS17.3.1无法通过快捷指令上传照片（通过相册分享到快捷指令），提示图片上传失败是否继续，请问有无微调修复的方法

已修复

**Comment by gzxiexl** [comment]
好像最新版又不能用了🤦‍♂️

**Reply by XoYaPeng** [reply]
不能用了，等更新？

**Comment by lllllpb** [comment]
图片上传失败，作者大佬求更新

**Comment by swiftwind0405** [comment]
求更新啊
